File: whatsapp-backend/src/main/java/com/ah/whatsapp/constant/WebSocketConstants.java
package com.ah.whatsapp.constant;

public class WebSocketConstants {
	private WebSocketConstants() {
    }

	public static final String APP_PREFIX = "/app";
    public static final String CHAT_SEND_MESSAGE = "/chat.sendMessage"; // Relative to APP_PREFIX

    // Broker Destinations (Server -> Client @SendTo / SimpMessagingTemplate)
    public static final String TOPIC_PREFIX = "/topic";
    public static final String CONVERSATIONS_TOPIC = "/conversations"; // Relative to TOPIC_PREFIX
    public static final String CONVERSATION_TOPIC_TEMPLATE = TOPIC_PREFIX + CONVERSATIONS_TOPIC + "/%s"; // Template for specific conversation

    // STOMP Endpoint (Configured in WebSocketConfig, but useful as a constant)
    // Note: This includes the context path '/api' as configured
    public static final String STOMP_ENDPOINT = "/api/ws";
}

File: whatsapp-web/src/app/shared/constants/websocket.constants.ts
// Path for the initial SockJS HTTP connection (relative to environment.apiUrl)
export const WS_ENDPOINT_PATH = '/ws';

// Prefix for messages sent from client to server (@MessageMapping)
export const APP_PREFIX = '/app';

// Prefix for topics subscribed to by the client (Broker destinations)
export const TOPIC_PREFIX = '/topic';

// Specific destination for sending chat messages
export const CHAT_SEND_MESSAGE_DESTINATION = `${APP_PREFIX}/chat.sendMessage`;

// Function to generate the specific topic destination for a conversation
export const getConversationTopicDestination = (conversationId: string): string => {
  return `${TOPIC_PREFIX}/conversations/${conversationId}`;
};

export const USER_CONVERSATION_QUEUE = '/user/queue/conversations';
